src/dbus/server/dbus_thread_object_rcp.cpp

```diff
@@ -2010,7 +2010,16 @@ void DBusThreadObjectRcp::DeactivateEphemeralKeyModeHandler(DBusRequest &aReques
     otError error        = OT_ERROR_NONE;
     auto    threadHelper = mHost.GetThreadHelper();
 
+    VerifyOrExit(mBorderAgent.GetEphemeralKeyEnabled(), error = OT_ERROR_NOT_CAPABLE);
+
+    VerifyOrExit(otBorderAgentGetState(threadHelper->GetInstance()) == OT_BORDER_AGENT_STATE_STOPPED,
```

Comment:
shall we verify the ba state is not stopped here?

Reply:
Also, it may be simpler to just verify the state equals to `OT_BORDER_AGENT_STATE_ACTIVE`?

Reply:
The idea was to differentiate the case when ba is stopped and ba is active and return different error codes for them to provide more info to the client.

Reply:
I see. Probably we should add more test cases for this feature.

Currently the CI doesn't fail even when the condition is inverse, which is kind of concerning.

Reply:
CI does fail

https://github.com/openthread/ot-br-posix/actions/runs/10936291685/job/30386414875?pr=2496#step:9:594
```
2024-09-19 16:08:00,915 - INFO - Otbr<1> $ python3 /app/third_party/openthread/repo/tests/scripts/thread-cert/call_dbus_method.py io.openthread.BorderRouter DeactivateEphemeralKeyMode '[]'
Traceback (most recent call last):
  File "/app/third_party/openthread/repo/tests/scripts/thread-cert/call_dbus_method.py", line 46, in <module>
    main()
  File "/app/third_party/openthread/repo/tests/scripts/thread-cert/call_dbus_method.py", line 41, in main
    res = method(*arguments)
  File "/usr/lib/python3/dist-packages/dbus/proxies.py", line 70, in __call__
    return self._proxy_method(*args, **keywords)
  File "/usr/lib/python3/dist-packages/dbus/proxies.py", line 145, in __call__
    **keywords)
  File "/usr/lib/python3/dist-packages/dbus/connection.py", line 651, in call_blocking
    message, timeout)
dbus.exceptions.DBusException: io.openthread.Error.Failed: io.openthread.Error.Failed
```

Reply:
may consider using `switch(otBorderAgentGetState(threadHelper->GetInstance())` instead

Reply:
Thanks! Modified accordingly. reopen if improper.